frontend/src/components/normal-playlist/playlist.js

```diff
@@ -0,0 +1,253 @@
+    spotifyApi.addTracksToPlaylist(playlistId, topTracksIds).then(
+        (data) => {
+          //this.resetIframe()
+        },
+        (err) => {
+            console.log(err)
+        }
+    )
+
+    this.setState({
+        topTracksIds: []
+    })
+  }
+
+  handleSelect = (value) => {
+    this.setState({
+        selectedArtists: this.state.selectedArtists.concat(value)
+    }, () => {
+        console.log(this.state.selectedArtists)
+    }) 
+  }
+
+  handleDeselect = (value) => {
+    this.setState({
+        selectedArtists: this.state.selectedArtists.filter(name => !name.includes(value))
+    }, () => {
+        console.log(this.state.selectedArtists)
+    });
+
+
+  }
+
+  handleCreate = () => {
+    this.createPlaylist(this.state.user.id, this.state.playlistName, this.state.playlistDescription);
+    this.setState({
+        displayPlaylist: 'block',
+      })
+
+    setTimeout(() => { // iframe needs a reload to show songs
+        this.setState({ random: this.state.random + 1 })
+    }, 700);
+
+    setTimeout(() => { // selected songs dissapear 
+        this.setState({
+            selectedArtists: [],
+            topTracksIds: []
+        })
+    },3000)
+  }
+
+  createPlaylist = (userId, name, description) => {
+    spotifyApi.createPlaylist(userId, {name, description}).then(
+      (data) => {
+
+          console.log('once' + this.state.user.id)
+          this.setState({
+              user: { id: userId ,playlistId: data.id, iframe: `https://open.spotify.com/embed/playlist/${data.id}` } 
+          }, () => {
+              this.searchArtist(this.state.selectedArtists)
+          })
+          
+      },
+      (err) => {
+          console.log(err)
+      }
+    )
+  }
+
+  handleNameInput = (event) => {
+    this.setState({
+        playlistName: event.target.value
+    })
+  }
+
+  handleDescriptionInput = (event) => {
+    this.setState({
+        playlistDescription: event.target.value
+    })
+  }
+
+  render() {
+    return (
+        <Container className="playlistPage-container">
+            {this.state.displayPlaylist !== 'none' ? <Row id="iframe-row"><EmbeddedPlaylist key={this.state.random} link={this.state.user.iframe}/></Row> : ''}
+            <Row id="playlistPage-row">
+                <Col>
+                    <div id="form-container">
+                        <div id="artistInput">
+                            <div id="searchArtist">
+                                <Search
+                                    id="searchArtistInput"
+                                    placeholder="Search Artist"
+                                    onSearch={ value => this.getArtist(value)}
+                                    enterButton
+                                />
+                            </div>
+                            <Button style={{display: this.state.selectedArtists.length > 0 ? 'block' : 'none'}} id ="create-button" type="primary" onClick={this.handleCreate}> Create Playlist </Button>
```

Comment:
This is still an issue as i was able to create a playlist without giving the playlist a name which results in an [error](https://imgur.com/mp8ZIwt) since it is a [required](https://developer.spotify.com/documentation/web-api/reference/playlists/create-playlist/) field for the api call. If a user enters a playlist name and then erases it, then tries to create the playlist, the playlist name will be empty.

I think what you could do is check if the playlistName state value has anything inside of it before the api call is done. If there is nothing inside of it then dont do the api call. 

If we pair that with the [form field](https://ant.design/components/form/) from antd then I think it will solve our issue.

Reply:
I want to let user decide if he wants to give a name and description to playlist. So I added this before but I think state value is getting lost at somewhere. 

`  constructor(props) {
    super(props);
    this.state={
      user: {
          id: '',
          playlistId: '',
          iframe:''
      },
      artist: {
          name: '',
          id: ''
      },
      playlistName: 'Playlist App',
      playlistDescription: 'Created with Playlist App',
      selectedArtists: [], 
      searchedArtists: [],
      topTracksIds: [],
      displayPlaylist: 'none',
      random: 0
    }
  }`

So for the problem I am adding this. Is it ok?

`  createPlaylist = (userId, name, description) => {
    if(name === undefined || description === undefined) {
        name = 'Playlist App';
        description= 'Created with Playlist App!'
    }`

When calling the method we are using states so I am not giving default value due possibility to of passing undefined.

Reply:
If everything is smooth I'd be happy you merge with fixed conflicts.